### PR TITLE
Feat/tsemach/add command for promote app

### PR DESCRIPTION
**Pull request by tsemachLi**
This PR adds a new `app:promote` command to the CLI.

#### **Features:**
- **App Selection:** Choose the app you want to promote.
- **Promotion Options:**
  - Promote the **latest draft** version.
  - Specify a **specific version** to promote.

### CODE REVIEW

File: src/services/app-promote-service.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    body: { ...(appVersionId && { appVersionId: appVersionId.toString() }) },
+  });
+  ctx.retryAfter = Number(response.headers['retry-after']) || DEFAULT_DELAY_POLLING_MS / 1000;
+  ctx.urlToPull = appsUrlBuilder(pullPromoteStatusUrl(response.headers.location as string));
+};
+
+const sleep = (ms: number): Promise<void> =>
+  new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
```
Comment (gregra81) [inline src/services/app-promote-service.ts:43]: extract this to a util file to be re-used

File: src/services/app-promote-service.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+const handleStatusUpdate = (
+  task: ListrTaskWrapper<PromoteCommandTasksContext, any>,
+  status: APP_VERSION_STATUS,
+): void => {
+  switch (status) {
+    case APP_VERSION_STATUS.DRAFT: {
+      task.title = 'Failed to promote the app, back to draft status';
```
Comment (gregra81) [inline src/services/app-promote-service.ts:60]:
```suggestion
      task.title = 'Failed to promote the app, rolling back to a draft version';
```

File: src/services/app-promote-service.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  return pollPromotionStatus(urlToPull, task);
+};
+
+export const pullPromoteStatusTask = async (
+  ctx: PromoteCommandTasksContext,
+  task: ListrTaskWrapper<PromoteCommandTasksContext, any>,
+): Promise<void> => {
+  const { urlToPull, retryAfter } = ctx;
+  await sleep(retryAfter! * 1000);
+  await pollPromotionStatus(urlToPull!, task);
```
Comment (gregra81) [inline src/services/app-promote-service.ts:96]: this can create issue, right? How do you know that retryAfter and urlToPull are not undefined?

**Review by gregra81 — COMMENTED** [review]
Nice! Can't believe this is coming to life
Left you a few comments

File: src/services/app-promote-service.ts (hunk 3 above)
Comment (tsemachLi) [reply src/services/app-promote-service.ts:96]: Goos point

**Review by gregra81 — APPROVED** [review]

**Pull request merged**